packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1033,6 +1033,16 @@ export interface BucketProps {
    */
   readonly encryptionKey?: kms.IKey;
 
+  /**
+   * Specifies whether Amazon S3 should use an S3 Bucket Key with server-side
+   * encryption using KMS (SSE-KMS) for new objects in the bucket.
+   *
+   * Only relevant, when Encryption is set to {@link BucketEncryption.KMS}
+   *
+   * @default - undefined
```

Comment:
I disagree. This parameter is completely irrelevant, when SSE is not KMS. Hence imho it does make more sense to leave `undefined` as default.

Reply:
Sure, but what if `SSE` is set to `KMS`, and this parameter is not specified, then `false` would be relevant. Again this would have to be `false` if we remove the distinction of `undefined` as I mention [here](https://github.com/aws/aws-cdk/pull/12376#discussion_r558301365)